conan/tools/cmake/cmakedeps/templates/__init__.py

```diff
@@ -25,8 +25,10 @@ def file_name(self):
         return self.cmakedeps.get_cmake_package_name(self.conanfile, module_mode=self.generating_module) + self.suffix
 
     @property
-    def legacy_variable_prefix(self):
-        return self.cmakedeps.get_property("cmake_legacy_variable_prefix", self.conanfile) or self.file_name
+    def additional_variables_prefixes(self):
+        prefix_list = self.cmakedeps.get_property("cmake_additional_variables_prefixes", self.conanfile)
+        prefix_list = prefix_list if prefix_list else []
+        return list(set([self.file_name] + prefix_list))
```

Comment:
you can simplify this a bit:

```suggestion
    def additional_variables_prefixes(self):
        prefix_list = self.cmakedeps.get_property("cmake_additional_variables_prefixes", 
                                                  self.conanfile) or []
        return list(set([self.file_name] + prefix_list))
```